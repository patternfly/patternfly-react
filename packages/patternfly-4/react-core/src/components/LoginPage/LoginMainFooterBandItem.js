import React from 'react';
import styles from '@patternfly/patternfly/components/Login/login.css';
import { css } from '@patternfly/react-styles';
import PropTypes from 'prop-types';

const propTypes = {
  /** Content rendered inside the footer Link Item */
  children: PropTypes.node,
  /** Additional classes added to the Footer Link Item  */
  className: PropTypes.string,
  /** Additional props are spread to the container <a> */
  '': PropTypes.any
};

const defaultProps = {
  children: null,
  className: ''
};

const LoginMainFooterBandItem = ({ className, children, ...props }) => (
  <p className={css(`${styles.loginMainFooterBand}-item`, className)} {...props}>
    {children}
  </p>
);

LoginMainFooterBandItem.propTypes = propTypes;
LoginMainFooterBandItem.defaultProps = defaultProps;

export default LoginMainFooterBandItem;
