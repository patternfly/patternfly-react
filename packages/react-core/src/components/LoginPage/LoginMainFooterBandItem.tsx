import * as React from 'react';
import styles from '@patternfly/react-styles/css/components/Login/login';
import { css } from '@patternfly/react-styles';

export interface LoginMainFooterBandItemProps extends React.HTMLProps<HTMLParagraphElement> {
  /** Content rendered inside the footer Link Item */
  children?: React.ReactNode;
  /** Additional classes added to the Footer Link Item  */
  className?: string;
}

export const LoginMainFooterBandItem: React.FunctionComponent<LoginMainFooterBandItemProps> = ({
  children = null,
  className = '',
  ...props
}: LoginMainFooterBandItemProps) => (
  <p className={css(`${styles.loginMainFooterBand}-item`, className)} {...props}>
    {children}
  </p>
);
LoginMainFooterBandItem.displayName = 'LoginMainFooterBandItem';
