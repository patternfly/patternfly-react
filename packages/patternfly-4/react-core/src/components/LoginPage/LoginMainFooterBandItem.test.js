import React from 'react';
import LoginMainFooterBandItem from './LoginMainFooterBandItem';
import { shallow } from 'enzyme';

test('renders with PatternFly Core styles', () => {
  const view = shallow(<LoginMainFooterBandItem />);
  expect(view).toMatchSnapshot();
});

test('className is added to the root element', () => {
  const view = shallow(<LoginMainFooterBandItem className="extra-class" />);
  expect(view.prop('className')).toMatchSnapshot();
});

test('extra props are spread to the root element', () => {
  const testId = 'login-body';
  const view = shallow(<LoginMainFooterBandItem data-testid={testId} />);
  expect(view.prop('data-testid')).toBe(testId);
});

test('LoginFooterItem  with custom node', () => {
  const CustomNode = () => <div>My custom node</div>;
  const view = shallow(
    <LoginMainFooterBandItem>
      <CustomNode />
    </LoginMainFooterBandItem>
  );
  expect(view).toMatchSnapshot();
});
