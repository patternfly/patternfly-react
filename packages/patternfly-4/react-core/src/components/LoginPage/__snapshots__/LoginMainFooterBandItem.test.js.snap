// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginFooterItem  with custom node 1`] = `
<p
  className="[object Object]-item"
>
  <CustomNode />
</p>
`;

exports[`className is added to the root element 1`] = `"[object Object]-item extra-class"`;

exports[`renders with PatternFly Core styles 1`] = `
<p
  className="[object Object]-item"
/>
`;
